### PR TITLE
 Use async/await for asynchronous calls

### PR DESCRIPTION
**Pull request by Mitch528**

### CODE REVIEW

**Comment by cozyplanes** [comment]
May I ask what has been changed? Also please resolve conflicts.

**Comment by Kimmax** [comment]
I approve (y)
Good work

**Comment by Kimmax** [comment]
The 56 character limit for the subtext is missing. Everything else is handled and can be merged

**Comment by Kimmax** [comment]
Uhm.. I resolved the conflict but github merged in the wrong direction...
Will have to take care of this from CLI

**Comment by Kimmax** [comment]
Second time worked. Lol

File: devRantBanner/Controllers/BannerController.cs [hunk 1]
```diff
@@ -12,26 +12,29 @@ namespace devBanner.Controllers
...
-            // Convert username to userID
             var client = DevRantClient.Create(new HttpClient());
-            var userId = client.GetUserID(username).Result.UserId;
+
+            // Convert username to userID
+            var userId = await client.GetUserID(username);
+
+            if (!userId.Success)
+            {
+                return BadRequest("User does not exist!");
```
Comment (Kimmax) [inline devRantBanner/Controllers/BannerController.cs]: Is BadRequest() built in?

Comment (cozyplanes) [reply devRantBanner/Controllers/BannerController.cs]: Seems like it

Is this related? https://msdn.microsoft.com/en-us/library/system.identitymodel.badrequestexception(v=vs.110).aspx

Comment (Kimmax) [reply devRantBanner/Controllers/BannerController.cs]: unrelated

Comment (Kimmax) [reply devRantBanner/Controllers/BannerController.cs]: [Controller](https://docs.microsoft.com/de-de/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.badrequest?view=aspnetcore-2.0) inherits from [ControllerBase](https://docs.microsoft.com/de-de/dotnet/api/Microsoft.AspNetCore.Mvc.ControllerBase?view=aspnetcore-2.0) which implements [BadRequest](https://docs.microsoft.com/de-de/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.badrequest?view=aspnetcore-2.0).
So we are all green. Merging.

**Pull request merged**